### PR TITLE
add hack/README documentation for non-standard execution path of functional tests

### LINKED ISSUE #385: Update hack/README.md to add steps for non make cluster-up environments
add some basic steps needed to setup a local-up-cluster.sh environment and all the components that need to be running to be able to successfully run functional tests

### PR DESCRIPTION
**Pull request by screeley44**
fixes #385

### CODE REVIEW

File: hack/README.md [hunk 1]
```diff
@@ -77,27 +78,62 @@ These may be passed to a target as `$ make VARIABLE=value target`
...
+#### Execute Alternative Environment Functional Tests
+
+If running in a non-standard environment such as Mac or Cloud where the *kubevirtci framework* is
+not supported, then you can follow the following example to run Functional Tests.
+
+1. Stand-up a Kubernetes cluster (local-up-cluster.sh/kubeadm/minikube/etc...)
+
+2. Clone or get the kubevirt/containerized-data-importer repo
+
+3. Run the CDI controller manifests
```
Comment (copejon) [inline hack/README.md]: A note about building the file host should be included also as it is/will be required for some of the functional tests.  I've got pr #394 up now to add a `make docker-functest-image` target

**Comment by jeffvance** [comment]
nice addition. Thanks!

**Comment by jeffvance** [comment]
@screeley44 since you have this pr open would you mind adding the following for me to clarify better on recent changes I made to this file?

Change:
```
test-functional: execute functional tests under `./tests/...`. Additional test flags can be passed to the test binary via the TEST_ARGS variable, see below for an example.
``` 
to
```
test-functional: execute functional tests under `./tests/...`. Additional test flags can be passed to the test binary via the TEST_ARGS variable, see below for an example and restrictions.
```
and change:
```
TEST_ARGS: A variable containing a list of additional ginkgo flags to be passed to functional tests. The string "--test-args=" must prefix the variable value. For example: `make TEST_ARGS="--test-args=-ginkgo.noColor=true" test-functional >& foo`
```
to
```
TEST_ARGS: A variable containing a list of additional ginkgo flags to be passed to functional tests. The string "--test-args=" must prefix the variable value. For example: make TEST_ARGS="--test-args=-ginkgo.noColor=true" test-functional >& foo`.
Note: the following extra flags are not supported in TEST_ARGS: 
   -master
   -cdi-namespace
   -kubeconfig
   -kubectl-path   
```

File: hack/README.md [hunk 2]
```diff
@@ -76,31 +78,85 @@ These may be passed to a target as `$ make VARIABLE=value target`
...
-- `TEST_ARGS`: A variable containing a list of additional ginkgo flags to be passed to functional tests. The string "--test-args=" must prefix the variable value. For example: `make TEST_ARGS="--test-args=-ginkgo.noColor=true" test-functional >& foo`
+- `TEST_ARGS`: A variable containing a list of additional ginkgo flags to be passed to functional tests. The string "--test-args=" must prefix the variable value. For example:
+
+             `make TEST_ARGS="--test-args=-ginkgo.noColor=true" test-functional >& foo`.
+
+  Note: the following extra flags are not supported in TEST_ARGS:
+     -master
+     -cdi-namespace
+     -kubeconfig
+     -kubectl-path
```
Comment (jeffvance) [inline hack/README.md]: When viewed formatted this list of variables is a single list with no commas, etc. I think it should be a bullet-ed list or just a comma separated sentence.

File: hack/README.md [hunk 3]
```diff
@@ -76,31 +78,85 @@ These may be passed to a target as `$ make VARIABLE=value target`
...
+
```
Comment (jeffvance) [inline hack/README.md]: when viewed formatted you get an extra blank line here.

**Pull request merged**